pkg/client/route/client.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package route
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/route"
+
+	"github.com/hyperledger/aries-framework-go/pkg/common/connectionstore"
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// ErrConnectionNotFound is returned when connection not found
+var ErrConnectionNotFound = errors.New("connection not found")
+
+// provider contains dependencies for the route protocol and is typically created by using aries.Context()
+type provider interface {
+	Service(id string) (interface{}, error)
+	StorageProvider() storage.Provider
+	TransientStorageProvider() storage.Provider
+}
+
+// Client enable access to route api
+type Client struct {
+	connectionStore *connectionstore.ConnectionRecorder
+	routeSvc        protocolService
+}
+
+// protocolService defines DID Exchange service.
+type protocolService interface {
+	// DIDComm service
+	service.Handler
+
+	// SendRequest send route request
+	SendRequest(myDID, theirDID string) (string, error)
+}
+
+// New return new instance of route client
+func New(ctx provider) (*Client, error) {
+	svc, err := ctx.Service(route.Coordination)
+	if err != nil {
+		return nil, err
+	}
+
+	routeSvc, ok := svc.(protocolService)
+	if !ok {
+		return nil, errors.New("cast service to route service failed")
+	}
+
+	connectionStore, err := connectionstore.NewConnectionRecorder(ctx)
```

Comment:
Do you need R/W access to connections in this protocol?
(or read only.)

Reply:
just read only

Reply:
Also L32

Reply:
change it to connection lookup